lib/AST/Util.cpp

```diff
@@ -316,6 +316,26 @@ clang::Expr *Clone(clang::ASTUnit &unit, clang::Expr *expr,
   return CHECK_NOTNULL(cloner.Visit(CHECK_NOTNULL(expr)));
 }
 
+static clang::Expr *StripParens(clang::Expr *expr) {
+  if (auto paren = clang::dyn_cast<clang::ParenExpr>(expr)) {
+    return paren->getSubExpr();
+  }
+  return expr;
+}
+
+static clang::Expr *ApplyLNot(rellic::ASTBuilder &ast, clang::Expr *expr) {
+  if (auto unop = clang::dyn_cast<clang::UnaryOperator>(expr)) {
+    if (unop->getOpcode() == clang::UO_LNot) {
+      return unop->getSubExpr();
+    }
+  }
+  return ast.CreateLNot(expr);
+}
+
+clang::Expr *Negate(rellic::ASTBuilder &ast, clang::Expr *expr) {
```

Comment:
Do we really need to do this outside of `ExprCombine` ? There should be rewriting rules for this already, right?

Reply:
Since we changed the replacement mechanism in `NCP` to be syntactical, in this situation
```
if(!(a == b)) {
  ...
} else {
  if(a == b) {
    ...
  }
}
```
the condition would not be propagated to the `else` branch. The conditions need to be stripped down to the "bare minimum"

Reply:
Got it!